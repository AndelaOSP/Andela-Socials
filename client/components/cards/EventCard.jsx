import React from 'react';
import PropTypes from 'prop-types';


const EventCard = (props) => {
  document.title = 'Events page';
  const {
    eventTitle, eventDate, eventTags,
  } = props;
  return (
    <div className="event-card">
      <img src="/assets/img/img_group_selfie.jpg" alt="event img" className="event-card__picture" />
      <div className="event-card__caption-group">
        <p className="event-card__caption event-card__caption--title">{eventTitle || 'Swimming Meetup'}</p>
        <p className="event-card__caption event-card__caption--date">{eventDate || 'Saturday, July 22nd 2018'}</p>
        <p className="event-card__caption event-card__caption--tags">{eventTags || '#swimming #meetup'}</p>
      </div>
    </div>

  );
};
EventCard.propTypes = {
  eventDate: PropTypes.string,
  eventTags: PropTypes.string,
  eventTitle: PropTypes.string,
};

EventCard.defaultProps = {
  eventDate: '',
  eventTags: '',
  eventTitle: '',
};


export default EventCard;
