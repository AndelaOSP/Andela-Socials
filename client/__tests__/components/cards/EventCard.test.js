/* eslint-env mocha */
import React from 'react';
import { shallow } from 'enzyme';

import EventCard from '../../../components/cards/EventCard';

describe('Search Modal', () => {
  it('renders', () => {
    const instance = shallow(<EventCard />);
    expect(instance).toMatchSnapshot();
  });
});
