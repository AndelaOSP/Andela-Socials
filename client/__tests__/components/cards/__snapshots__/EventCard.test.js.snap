// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Modal renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    featuredImage=""
    id=""
    socialEvent={Object {}}
    startDate=""
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="event-card"
      >
        <img
          alt=""
          className="event-card__picture"
          src="/assets/img/img_group_selfie.jpg"
        />
        <div
          className="event-card__caption-group"
        >
          <p
            className="event-card__caption event-card__caption--title"
          >
            Swimming Meetup
          </p>
          <p
            className="event-card__caption event-card__caption--date"
          >
            Invalid Date
          </p>
          <p
            className="event-card__caption event-card__caption--tags"
          >
            #undefined
          </p>
        </div>
      </div>,
      "className": "event-link",
      "href": "/events/",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className="event-card__picture"
            src="/assets/img/img_group_selfie.jpg"
          />,
          <div
            className="event-card__caption-group"
          >
            <p
              className="event-card__caption event-card__caption--title"
            >
              Swimming Meetup
            </p>
            <p
              className="event-card__caption event-card__caption--date"
            >
              Invalid Date
            </p>
            <p
              className="event-card__caption event-card__caption--tags"
            >
              #undefined
            </p>
          </div>,
        ],
        "className": "event-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "event-card__picture",
            "src": "/assets/img/img_group_selfie.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="event-card__caption event-card__caption--title"
              >
                Swimming Meetup
              </p>,
              <p
                className="event-card__caption event-card__caption--date"
              >
                Invalid Date
              </p>,
              <p
                className="event-card__caption event-card__caption--tags"
              >
                #undefined
              </p>,
            ],
            "className": "event-card__caption-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Swimming Meetup",
                "className": "event-card__caption event-card__caption--title",
              },
              "ref": null,
              "rendered": "Swimming Meetup",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Invalid Date",
                "className": "event-card__caption event-card__caption--date",
              },
              "ref": null,
              "rendered": "Invalid Date",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "#undefined",
                "className": "event-card__caption event-card__caption--tags",
              },
              "ref": null,
              "rendered": "#undefined",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="event-card"
        >
          <img
            alt=""
            className="event-card__picture"
            src="/assets/img/img_group_selfie.jpg"
          />
          <div
            className="event-card__caption-group"
          >
            <p
              className="event-card__caption event-card__caption--title"
            >
              Swimming Meetup
            </p>
            <p
              className="event-card__caption event-card__caption--date"
            >
              Invalid Date
            </p>
            <p
              className="event-card__caption event-card__caption--tags"
            >
              #undefined
            </p>
          </div>
        </div>,
        "className": "event-link",
        "href": "/events/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              className="event-card__picture"
              src="/assets/img/img_group_selfie.jpg"
            />,
            <div
              className="event-card__caption-group"
            >
              <p
                className="event-card__caption event-card__caption--title"
              >
                Swimming Meetup
              </p>
              <p
                className="event-card__caption event-card__caption--date"
              >
                Invalid Date
              </p>
              <p
                className="event-card__caption event-card__caption--tags"
              >
                #undefined
              </p>
            </div>,
          ],
          "className": "event-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "event-card__picture",
              "src": "/assets/img/img_group_selfie.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="event-card__caption event-card__caption--title"
                >
                  Swimming Meetup
                </p>,
                <p
                  className="event-card__caption event-card__caption--date"
                >
                  Invalid Date
                </p>,
                <p
                  className="event-card__caption event-card__caption--tags"
                >
                  #undefined
                </p>,
              ],
              "className": "event-card__caption-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Swimming Meetup",
                  "className": "event-card__caption event-card__caption--title",
                },
                "ref": null,
                "rendered": "Swimming Meetup",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Invalid Date",
                  "className": "event-card__caption event-card__caption--date",
                },
                "ref": null,
                "rendered": "Invalid Date",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "#undefined",
                  "className": "event-card__caption event-card__caption--tags",
                },
                "ref": null,
                "rendered": "#undefined",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
